### ISSUE TITLE
Multinode install fails. Error 6503 & HTTP ERROR 500 in UI.

### ISSUE THREAD

**Issue by dassic** [EMCECS/ECS-CommunityEdition#180]
Hi!

Have tried a couple of times with the multi-node install on CentOS 7.
All seems to go well on the 6 nodes I am installing on, but when I log into the UI after having finished step1 on all nodes I get a red error bar on the top stating Error 6503 for half a second and then HTTP ERROR: 500.....and just about nothing works in the UI.

Only thing during install that may indicate an error is this during the disk layout : 

create file 0196
create file 0197
field u not found
field u not found
field u not found

The "field u not found" is repeated many times.

I have checked that nothing else is running on port 80, 443 nor on 4443.

Single node install works fine.

Any suggestions to what is wrong ?

/Jesper

**Comment by dassic** [comment]
Looks like for multinode install it requires lots of additional ports, one of them being port 9011 which conflict with ScaleIO MDM, so I think this is the reason for the install to fail.

Is there a way to renumber 9011 to something else ?

**Comment by jasoncwik** [comment]
Hmm... there _might_ be a way to do this by editing ECS's configuration files.  Maybe we can spin a custom container for SIO machines.  Can you stop the ECS container and execute the following in the root OS so I can see the full port list?

`netstat -anp | grep LISTEN`

**Comment by dassic** [comment]
Here you go. It's on CentOS 7.

[root@ssfns-sds-server-1 ~]# netstat -anp | grep LISTEN
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      1/systemd           
tcp        0      0 192.168.122.1:53        0.0.0.0:*               LISTEN      3626/dnsmasq        
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      2066/sshd           
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      3798/master         
tcp        0      0 127.0.0.1:199           0.0.0.0:*               LISTEN      2018/snmpd          
tcp6       0      0 :::9099                 :::*                    LISTEN      2070/lia-2.0.11000. 
tcp6       0      0 :::111                  :::*                    LISTEN      1/systemd           
tcp6       0      0 :::8080                 :::*                    LISTEN      2062/java           
tcp6       0      0 :::6611                 :::*                    LISTEN      38522/mdm-2.0.11000 
tcp6       0      0 :::9011                 :::*                    LISTEN      38522/mdm-2.0.11000 
tcp6       0      0 :::25620                :::*                    LISTEN      38522/mdm-2.0.11000 
tcp6       0      0 :::22                   :::*                    LISTEN      2066/sshd           
tcp6       0      0 ::1:25                  :::*                    LISTEN      3798/master         
tcp6       0      0 :::8443                 :::*                    LISTEN      2062/java           
tcp6       0      0 :::7072                 :::*                    LISTEN      2015/sds-2.0.11000. 
tcp6       0      0 :::25640                :::*                    LISTEN      2015/sds-2.0.11000. 
unix  2      [ ACC ]     STREAM     LISTENING     28052    3798/master          private/bounce
unix  2      [ ACC ]     STREAM     LISTENING     1684501  139554/docker-curre  /var/run/docker.sock
unix  2      [ ACC ]     STREAM     LISTENING     28055    3798/master          private/defer
unix  2      [ ACC ]     STREAM     LISTENING     28058    3798/master          private/trace
unix  2      [ ACC ]     STREAM     LISTENING     28061    3798/master          private/verify
unix  2      [ ACC ]     STREAM     LISTENING     28067    3798/master          private/proxymap
unix  2      [ ACC ]     STREAM     LISTENING     16410    1/systemd            /run/systemd/journal/stdout
unix  2      [ ACC ]     STREAM     LISTENING     28070    3798/master          private/proxywrite
unix  2      [ ACC ]     STREAM     LISTENING     28091    3798/master          private/local
unix  2      [ ACC ]     STREAM     LISTENING     13854    1206/gssproxy        /run/gssproxy.sock
unix  2      [ ACC ]     STREAM     LISTENING     1677831  139554/docker-curre  /var/lib/docker/network/files/89b62c04a120a0895261f4b6f29ddaf1d3099800a1de38e0ef2ef7c90f124ab6.sock
unix  2      [ ACC ]     STREAM     LISTENING     28094    3798/master          private/virtual
unix  2      [ ACC ]     STREAM     LISTENING     28088    3798/master          private/discard
unix  2      [ ACC ]     STREAM     LISTENING     28076    3798/master          private/relay
unix  2      [ ACC ]     STREAM     LISTENING     28082    3798/master          private/error
unix  2      [ ACC ]     STREAM     LISTENING     33143    1/systemd            @ISCSID_UIP_ABSTRACT_NAMESPACE
unix  2      [ ACC ]     SEQPACKET  LISTENING     45132    1395/teamd           /var/run/teamd/team0.sock
unix  2      [ ACC ]     STREAM     LISTENING     33870    1/systemd            /run/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     29785    1/systemd            /run/lvm/lvmpolld.socket
unix  2      [ ACC ]     STREAM     LISTENING     28073    3798/master          private/smtp
unix  2      [ ACC ]     STREAM     LISTENING     28042    3798/master          public/qmgr
unix  2      [ ACC ]     SEQPACKET  LISTENING     15469    1/systemd            /run/udev/control
unix  2      [ ACC ]     STREAM     LISTENING     28035    3798/master          public/pickup
unix  2      [ ACC ]     STREAM     LISTENING     33139    1/systemd            /var/run/libvirt/virtlogd-sock
unix  2      [ ACC ]     STREAM     LISTENING     28039    3798/master          public/cleanup
unix  2      [ ACC ]     STREAM     LISTENING     28097    3798/master          private/lmtp
unix  2      [ ACC ]     STREAM     LISTENING     33144    1/systemd            /var/run/rpcbind.sock
unix  2      [ ACC ]     STREAM     LISTENING     32891    1/systemd            /run/lvm/lvmetad.socket
unix  2      [ ACC ]     STREAM     LISTENING     33148    1/systemd            /var/run/dbus/system_bus_socket
unix  2      [ ACC ]     STREAM     LISTENING     28085    3798/master          private/retry
unix  2      [ ACC ]     STREAM     LISTENING     28100    3798/master          private/anvil
unix  2      [ ACC ]     STREAM     LISTENING     28103    3798/master          private/scache
unix  2      [ ACC ]     STREAM     LISTENING     13853    1206/gssproxy        /var/lib/gssproxy/default.sock
unix  2      [ ACC ]     STREAM     LISTENING     15522    1194/abrtd           /var/run/abrt/abrt.socket
unix  2      [ ACC ]     STREAM     LISTENING     21933    1201/lsmd            /var/run/lsm/ipc/sim
unix  2      [ ACC ]     STREAM     LISTENING     21935    1201/lsmd            /var/run/lsm/ipc/simc
unix  2      [ ACC ]     STREAM     LISTENING     32186    2086/libvirtd        /var/run/libvirt/libvirt-sock
unix  2      [ ACC ]     STREAM     LISTENING     32188    2086/libvirtd        /var/run/libvirt/libvirt-sock-ro
unix  2      [ ACC ]     STREAM     LISTENING     32190    2086/libvirtd        /var/run/libvirt/libvirt-admin-sock
unix  2      [ ACC ]     STREAM     LISTENING     33142    1/systemd            @ISCSIADM_ABSTRACT_NAMESPACE
unix  2      [ ACC ]     STREAM     LISTENING     28046    3798/master          private/tlsmgr
unix  2      [ ACC ]     STREAM     LISTENING     28049    3798/master          private/rewrite
unix  2      [ ACC ]     STREAM     LISTENING     28064    3798/master          public/flush
unix  2      [ ACC ]     STREAM     LISTENING     28079    3798/master          public/showq

**Comment by jasoncwik** [comment]
OK, so SIO is using these ports: 

```
tcp6 0 0 :::9099 :::* LISTEN 2070/lia-2.0.11000.
tcp6 0 0 :::6611 :::* LISTEN 38522/mdm-2.0.11000
tcp6 0 0 :::9011 :::* LISTEN 38522/mdm-2.0.11000
tcp6 0 0 :::25620 :::* LISTEN 38522/mdm-2.0.11000
tcp6 0 0 :::7072 :::* LISTEN 2015/sds-2.0.11000.
tcp6 0 0 :::25640 :::* LISTEN 2015/sds-2.0.11000.
```

By inspection, I think we have conflicts on both 9011 and 9099:

```
ecs-obj-1-1:/ # netstat -anp | grep LISTEN | grep tcp | grep 9011
tcp        0      0 127.0.0.1:9011          :::*                    LISTEN      12091/objcontrolsvc
ecs-obj-1-1:/ # netstat -anp | grep LISTEN | grep tcp | grep 9099
tcp        0      0 10.1.83.51:9099         0.0.0.0:*               LISTEN      12073/storageserver
```
storageserver might be hard to move.  Let me see what I can come up with.

**Comment by jasoncwik** [comment]
@dassic unfortunately, I we don't have time to develop a workaround for this right now.  You could try the following:

1) enter the container
2) change directory to /opt/storageos/conf and change all instances of 9011 and 9099 to something else.  Do the same on all other nodes.
3) reboot all the nodes

That might work, but no guarantees.

**Comment by padthaitofuhot** [comment]
Assigning @captntuttle to document popular port conflicts.

**Comment by padthaitofuhot** [comment]
The installer now detects and fails with an informative error message (which port is conflicting) when ECS ports are in use.  Addressed by #179 and #226

**Comment by padthaitofuhot** [comment]
@dassic Please reopen this issue if we can be of further assistance.  Thank you.
